### ISSUE TITLE
[DataTable] A11Y: The first column header does not receive focus / skips to second column header

### ISSUE THREAD

**Issue by ginabak** [Shopify/polaris-react-archive#5272]
## Issue summary

The [Data Table Component](https://polaris.shopify.com/components/lists-and-tables/data-table)'s first column header does not receive focus / skips to second column header

https://user-images.githubusercontent.com/43223543/157984429-5a597a0a-4a7b-4a2f-aa0f-6e946152dcc1.mp4

## Expected behavior

The [Data Table Component](https://polaris.shopify.com/components/lists-and-tables/data-table)'s first column header should receive focus and not skip to the second column header

## Steps to reproduce the problem

This [CodeSandbox template](https://lxswwr.csb.app/) contains the Data Table Polaris component. 
1. Enable VoiceOver (command F5) 
2. Tab through columns via the keyboard using `control` `option` arrow key! 
3. Notice that the first focusable header is the second column header rather than the first 

Same component issues https://github.com/Shopify/polaris-react/issues/5268 https://github.com/Shopify/polaris-react/issues/5270
